front-end/src/app/shared/components/poller/poller.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { PollerService } from 'app/shared/services/poller.service';
+import { Subscription } from 'rxjs';
+import { Location } from '@angular/common';
+import { ConfirmationService } from 'primeng/api';
+
+@Component({
+  selector: 'app-poller',
+  templateUrl: './poller.component.html',
+  styleUrl: './poller.component.scss',
+})
+export class PollerComponent implements OnInit, OnDestroy {
+  isNewVersionAvailable = false;
+  private subscription!: Subscription;
+
+  constructor(
+    private pollerService: PollerService,
+    private location: Location,
+    private confirmationService: ConfirmationService,
+  ) {}
+
+  ngOnInit() {
+    const currentUrl = window.location.href;
+    const index = currentUrl.indexOf(this.location.path());
+
+    let deploymentUrl = index === 0 ? currentUrl : currentUrl.substring(0, index);
+    deploymentUrl += deploymentUrl.endsWith('/') ? 'index.html' : '/index.html';
+
+    this.pollerService.startPolling(deploymentUrl);
+    this.subscription = this.pollerService.isNewVersionAvailable$.subscribe((isNex) => {
+      this.isNewVersionAvailable = isNex;
+      if (this.isNewVersionAvailable) {
+        this.confirmationService.confirm({
+          key: 'refresh',
+          message:
+            'A new version of the application has been deployed! Would you like to refresh the page to receive the new app?',
```

Comment:
Good catch that this could be jarring for a user if they're in the middle of something.  This ticket is only to refresh when the versions don't match.  We're going to need to take this confirmation out or consult with design to get this part added to the ticket.